src/core/link.lua

```diff
@@ -2,11 +2,16 @@ module(...,package.seeall)
 
 local debug = _G.developer_debug
 
+local shm = require("core.shm")
 local ffi = require("ffi")
 local C = ffi.C
 
 local packet = require("core.packet")
 require("core.packet_h")
+
+local counter = require("core.counter")
+require("core.counter_h")
```

Comment:
I copied the pattern from the `core.packet_h` require just above. Indeed the requires of both `packet_h` and `counter_h` are redundant. I suggest we fix this in a separate "style branch", since we have a **lot** of unnecessary `require` calls (especially to core packages) in most modules.